### PR TITLE
Refactor the ModuleStore to use ModuleEnhancer

### PR DESCRIPTION
**Pull request by navneet-g**

### CODE REVIEW

**Comment by navneet-g** [comment]
#24

File: README.md [hunk 1]
```diff
@@ -64,10 +64,10 @@ export function getUsersModule(): IModule<IUserState> {
 * Create a `ModuleStore`
 
 ```typescript
-import {configureStore, IModuleStore} from "redux-dynamic-modules";
+import {createStore, IModuleStore} from "redux-dynamic-modules";
```
Comment (abettadapur) [inline README.md]: Shouldn't this be from redux now?

File: packages/redux-dynamic-modules/src/ModuleEnhancer.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            // create module manager
+            const moduleManager = getRefCountedManager(
+                getModuleManager<any>(
+                    middlewareManager,
+                    extensions),
+                (a: IModule<any>, b: IModule<any>) => a.id === b.id);
+
+            // Create module enhancer to manage extensions and dynamic middlewares
+            const moduleEnhancer = composeEnhancers(
+                applyMiddleware(...extensionMiddlewares,
```
Comment (abettadapur) [inline packages/redux-dynamic-modules/src/ModuleEnhancer.ts]: nit: newlines

File: packages/redux-dynamic-modules/src/ModuleEnhancer.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                return moduleManager.getReducer(intermediateState, action);
+            };
+
+            // Create the store
+            const store = createStore(chainedReducer, preloadedState, composedEnhancer);
+
+            // module manager will use the dispatch from store to dispatch initial and final actions
+            moduleManager.setDispatch(store.dispatch);
+
+            // adds given modules to mouldManager
```
Comment (abettadapur) [inline packages/redux-dynamic-modules/src/ModuleEnhancer.ts]: nit typo

File: packages/redux-dynamic-modules/src/ModuleStore.ts [hunk 4]
```diff
@@ -1,95 +1,23 @@
...
+export function createStore<S1, S2, S3, S4, S5, S6, S7>(initialState: DeepPartial<S1 & S2 & S3 & S4 & S5 & S6 & S7>, extensions: IExtension[], m1: IModule<S1>, m2: IModule<S2>, m3: IModule<S3>, m4: IModule<S4>, m5: IModule<S5>, m6: IModule<S6>, m7: IModule<S7>): IModuleStore<S1 & S2 & S3 & S4 & S5 & S6 & S7>;
+export function createStore<S1, S2, S3, S4, S5, S6, S7, S8>(initialState: DeepPartial<S1 & S2 & S3 & S4 & S5 & S6 & S7 & S8>, extensions: IExtension[], m1: IModule<S1>, m2: IModule<S2>, m3: IModule<S3>, m4: IModule<S4>, m5: IModule<S5>, m6: IModule<S6>, m7: IModule<S7>, m8: IModule<S8>): IModuleStore<S1 & S2 & S3 & S4 & S5 & S6 & S7 & S8>;
+export function createStore<State>(initialState: DeepPartial<State>, extensions: IExtension[], ...initialModules: IModule<any>[]): IModuleStore<State>;
+export function createStore<State>(initialState: DeepPartial<State>, extensions: IExtension[], ...initialModules: IModule<any>[]): IModuleStore<State> {
+    return reduxCreateStore(
+        /* reducer */undefined,
+        initialState,
+        moduleEnhancer(extensions, initialModules)
```
Comment (abettadapur) [inline packages/redux-dynamic-modules/src/ModuleStore.ts]: dynamicModuleEnhancer?

File: README.md (hunk 1 above)
Comment (abettadapur) [reply README.md]: And in all of the examples

Comment (navneetg-msft) [reply README.md]: createStore is a helper that we wrote for quickly using Module store with enhancer, so the enhancer is not exposed when not needed.

File: packages/redux-dynamic-modules/src/ModuleStore.ts (hunk 4 above)
Comment (navneetg-msft) [reply packages/redux-dynamic-modules/src/ModuleStore.ts]: I like moduleEnhancer better, it is short and clean.

File: packages/redux-dynamic-modules/src/ModuleEnhancer.ts (hunk 3 above)
Comment (navneetg-msft) [reply packages/redux-dynamic-modules/src/ModuleEnhancer.ts]: Will fix it later

**Pull request merged**